runtime/src/certify_key_extended.rs

```diff
@@ -0,0 +1,121 @@
+            let pdata = drivers.persistent_data.get_mut();
+            let crypto = DpeCrypto::new(
+                &mut drivers.sha384,
+                &mut drivers.trng,
+                &mut drivers.ecc384,
+                &mut drivers.hmac384,
+                &mut drivers.key_vault,
+                &mut pdata.fht.rt_dice_pub_key,
+                key_id_rt_cdi,
+                key_id_rt_priv_key,
+            );
+            let pl0_pauser = pdata.manifest1.header.pl0_pauser;
+            let (nb, nf) = Drivers::get_cert_validity_info(&pdata.manifest1);
+            let flags = CertifyKeyExtendedFlags(cmd.flags);
+            // Populate the otherName only if requested and provided by ADD_SUBJECT_ALT_NAME
+            let dmtf_device_info = if flags.contains(CertifyKeyExtendedFlags::DMTF_OTHER_NAME) {
+                drivers
+                    .dmtf_device_info
+                    .as_ref()
+                    .map(|dmtf_device_info| dmtf_device_info.as_bytes())
+            } else {
+                None
+            };
+            let mut env = DpeEnv::<CptraDpeTypes> {
+                crypto,
+                platform: DpePlatform::new(
+                    pl0_pauser,
+                    &hashed_rt_pub_key,
+                    &drivers.cert_chain,
+                    &nb,
+                    &nf,
+                    dmtf_device_info,
+                ),
+            };
+
+            let locality = drivers.mbox.user();
+            // Cannot call CERTIFY_KEY_EXTENDED from PL1
+            if Drivers::is_caller_pl1(pl0_pauser, pdata.manifest1.header.flags, locality) {
+                return Err(CaliptraError::RUNTIME_INCORRECT_PAUSER_PRIVILEGE_LEVEL);
+            }
+
+            let mut dpe = &mut pdata.dpe;
+            let certify_key_cmd = CertifyKeyCmd::read_from(&cmd.certify_key_req[..])
+                .ok_or(CaliptraError::RUNTIME_DPE_COMMAND_DESERIALIZATION_FAILED)?;
+            let resp = certify_key_cmd.execute(dpe, &mut env, locality);
+
+            let certify_key_resp = match resp {
+                Ok(Response::CertifyKey(certify_key_resp)) => certify_key_resp,
+                Ok(_) => return Err(CaliptraError::RUNTIME_CERTIFY_KEY_EXTENDED_FAILED),
+                Err(e) => {
+                    // If there is extended error info, populate CPTRA_FW_EXTENDED_ERROR_INFO
+                    if let Some(ext_err) = e.get_error_detail() {
+                        drivers.soc_ifc.set_fw_extended_error(ext_err);
+                    }
+                    return Err(CaliptraError::RUNTIME_CERTIFY_KEY_EXTENDED_FAILED);
+                }
+            };
+
+            let certify_key_extended_resp = CertifyKeyExtendedResp {
+                hdr: MailboxRespHeader::default(),
+                certify_key_resp: certify_key_resp
+                    .as_bytes()
+                    .try_into()
+                    .map_err(|_| CaliptraError::RUNTIME_CERTIFY_KEY_EXTENDED_FAILED)?,
```

Comment:
nit: Should we use a different error code for a serialization error?

Reply:
Done